pom.xml

```diff
@@ -208,7 +208,7 @@
     <pmd.version>7.18.0</pmd.version>
     <maven.jacoco.plugin.version>0.8.14</maven.jacoco.plugin.version>
     <mockito.version>5.2.0</mockito.version>
-    <slf4j.version>2.0.17</slf4j.version>
+    <slf4j.version>1.7.32</slf4j.version> <!-- Aligned with the version reflections depends on -->
```

Comment:
Historical context:

https://github.com/checkstyle/checkstyle/commit/981896d54ed2b1e7530c2f7d0e026c315f895ffa#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R264

https://github.com/checkstyle/checkstyle/issues/8742

This deprecated library was used to generate our internal resources during build time, it is not in use in any user visible functionality, so not required in runtime.

So better fix will be to exclude slf4j from such library and use it explicitly or take to it from other transitively.

https://github.com/ronmamo/reflections
https://mvnrepository.com/artifact/org.reflections/reflections

Reply:
Sorry, but I'm not sure I understand this comment.

Unless I'm missing something, this library *is* used at runtime, when using the `-f sarif` flag.
Usage chain:
`com.puppycrawl.tools.checkstyle.Main.OutputFormat` instantiates a new `com.puppycrawl.tools.checkstyle.SarifLogger`.
`com.puppycrawl.tools.checkstyle.SarifLogger#loadModuleMetadata` calls `com.puppycrawl.tools.checkstyle.meta.XmlMetaReader#readAllModulesIncludingThirdPartyIfAny`.
`com.puppycrawl.tools.checkstyle.meta.XmlMetaReader#readAllModulesIncludingThirdPartyIfAny` instantiates a new `org.reflections.Reflections` instance.
`org.reflections.Reflections`'s class loading instantiates a static `org.slf4j.Logger` member.

So, IIUC, excluding slf4j from reflections' dependencies won't work.

What am I missing?